src/luxonis_ml/embeddings/emb_OOD.py

```diff
@@ -0,0 +1,99 @@
+"""
+Out-of-Distribution Detection for Embeddings
+
+This module provides two primary methods for detecting out-of-distribution (OOD) samples 
+in embeddings. OOD samples can be crucial to identify as they represent anomalies or novel 
+patterns that don't conform to the expected distribution of the dataset.
+
+Methods available:
+- Isolation Forests: A tree-based model that partitions the space in such a manner that 
+  anomalies are isolated from the rest.
+  
+- Leverage with Linear Regression: Leverages (or hat values) represent the distance between 
+  the predicted values and the true values. Higher leverages indicate potential OOD points.
+
+Typical use cases include:
+- Anomaly Detection: Identifying rare patterns or outliers.
+  
+- Dataset Reduction: By removing or studying OOD samples, we can have a more homogeneous dataset.
+  
+- Expanding Datasets: Recognizing valuable data points that are distinct from the current distribution can be helpful 
+  when we're looking to diversify the dataset, especially in iterative learning scenarios.
+
+Dependencies:
+- numpy
+- scikit-learn
+
+"""
+
+import numpy as np
+from sklearn.ensemble import IsolationForest
+from sklearn.linear_model import LinearRegression
+
+def isolation_forest_OOD(X, contamination='auto', n_jobs=-1, verbose=1, random_state=42):
+    """
+    Out-of-distribution detection using Isolation Forests.
+
+    Parameters
+    ----------
+    X : np.array
+        The embeddings to use.
+    contamination : float
+        The contamination parameter for Isolation Forests. Default is 'auto'.
+    n_jobs : int
+        The number of jobs to use. Default is -1, which means all available CPUs.
+    verbose : int
+        The verbosity level. Default is 1.
+    random_state : int
+        The random state to use. Default is 42.
```

Comment:
Should the default `random_state` be `None`? I'm not sure. I could see it being nice if the default behavior is such that the result is reproducible.

Reply:
Yes I would set this to none and leave it to the user if they want it reproducible.